### PR TITLE
Language about security risks

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
#55

### CODE REVIEW

File: trace_context/security.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+considered as sensitive. For example, `traceparent` may allow to correlate calls so identify from one call may be correlated
+to the data sent with another call. `tracestate` may imply the version of monitored software used by caller. This
+information can be used to author an attack to the service.
+
+Even though the risk is minimal, application owner should make sure to control the boundary of organization's administrative
+control. Tracing vendors and platforms should allow to control any unintentional propagation of distributed tracing context.
+
+## Denial of monitoring by malicious caller
+
+When distributed tracing was enabled on a service with a public API and naively continue any trace with
```
Comment (yurishkuro) [inline trace_context/security.md]: s/was/is/

File: trace_context/security.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Security
+
+There are two areas of security risk - information exposure to malicious dependency service and denial of monitoring
+by malicious caller. Both risks exist with modern approaches for distributed trace context propagation. Standard makes
+it more likely for services to get these attacks.
```
Comment (yurishkuro) [inline trace_context/security.md]: why does it make it more likely?  The exact same risks exist today. Obscurity is not security.

File: trace_context/security.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+## Information exposure
+
+As mentioned in privacy section, information in `traceparent` and `tracestate` headers may carry information that can be
+considered as sensitive. For example, `traceparent` may allow to correlate calls so identify from one call may be correlated
+to the data sent with another call. `tracestate` may imply the version of monitored software used by caller. This
+information can be used to author an attack to the service.
+
+Even though the risk is minimal, application owner should make sure to control the boundary of organization's administrative
```
Comment (yurishkuro) [inline trace_context/security.md]: > Even though the risk is minimal

is an opinion, why do we need to state it?

File: trace_context/security.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+control. Tracing vendors and platforms should allow to control any unintentional propagation of distributed tracing context.
```
Comment (yurishkuro) [inline trace_context/security.md]: "control ... administrative  control" - could you please restate this more clearly?

File: trace_context/security.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+## Denial of monitoring by malicious caller
+
+When distributed tracing was enabled on a service with a public API and naively continue any trace with
+the sampling flag set using the `trace-id` provided via `traceparent` field, a malicious attacker could overwhelm an
```
Comment (yurishkuro) [inline trace_context/security.md]: "set using the `trace-id`" not necessary (makes it hard to read)

File: trace_context/security.md (hunk 1 above)
Comment (yurishkuro) [reply trace_context/security.md]: s/naively continue/naively continues/

File: trace_context/security.md (hunk 2 above)
Comment (SergeyKanzhelev) [reply trace_context/security.md]: The point here is that it is more likely to have generic attacks for standard headers than special attack for specialized `B2` or .NET headers. Like everybody tries to access wordpress admin portal to every domain without even checking. Do you think I can express this though clearer?

Comment (mtwo) [reply trace_context/security.md]: Perhaps we could reword the last sentence to "Having a standard for context propagation provides attackers with a known way of triggering tracing within an application"

File: trace_context/security.md (hunk 3 above)
Comment (mtwo) [reply trace_context/security.md]: Agreed, though I don't think it's a big deal. We can remove "Even though the risk is minimal"

Comment (mtwo) [reply trace_context/security.md]: Also, "application owner" should be changed to "application owners"

Comment (mtwo) [reply trace_context/security.md]: Let's change the paragraph to "Application owners should either ensure that no proprietary or confidential information is stored into the tracestate, or they should ensure that tracestate isn't present in requests to external systems".

**Review by mtwo — CHANGES_REQUESTED** [review]
I made some comments requesting minor changes, looks good otherwise

File: trace_context/security.md [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+the sampling flag set, a malicious attacker could overwhelm an application with tracing overhead, forge trace ID collisions
+that make monitoring data unusable, or run up your tracing bill with your SaaS tracing vendor.
+
+Tracing vendors and platforms should account to these situations and make sure that checks and balances are in place to
+protect denial of monitoring by malicious or badly-authored callers.
+
+## Other risks
+
+Application owner need to make sure to test all code path leading to sending of `traceparent` and `tracestate` headers. For
+example, browsers single page applications it is typical to make cross-origin calls from the browser. If one of code path leads
```
Comment (yurishkuro) [inline trace_context/security.md]: > browsers single page applications it is typical

something's off here

Comment (yurishkuro) [inline trace_context/security.md]: > If one of code path leads

paths

**Comment by mtwo** [comment]
I went through the security.md file and updated the grammar

**Review by mtwo — APPROVED** [review]

**Pull request merged**